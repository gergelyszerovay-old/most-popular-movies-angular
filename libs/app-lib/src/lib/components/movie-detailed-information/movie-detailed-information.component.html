<div *ngIf="movie; then isMovieSelected else noMovieSelected"></div>
<ng-template #isMovieSelected>
  <div class="c-detailed-info x-movie-selected">
    <button (click)="onCloseClick.emit()" aria-label="Close" class="close" type="button">
      <span aria-hidden="true">&times;</span>
    </button>
    <h3>{{movie.title}} ({{movie?.release_date | getYearFromISODate}})</h3>

    <ul class="c-detailed-info--genres">
      <li *ngFor="let genre of movie._genres; trackBy: trackByFnGenreList ">
        <span class="badge badge-secondary">{{genre.name}}</span>
      </li>
    </ul>
    <div class="c-detailed-info--rating">
      Rating:
      <ngb-rating [rate]="movie.vote_average" [readonly]="true"></ngb-rating>
      {{movie.vote_average | numberToFixed:1}}
    </div>
    <div class="row c-detailed-info--poster-overview">
      <img *ngIf="!!movie.poster_path" [src]="baseUrlPoster + movie.poster_path" class="col-md-4"/>
      <p class="col-md-8">{{movie.overview}}</p>
    </div>
  </div>
</ng-template>
<ng-template #noMovieSelected>
  <ngb-alert [dismissible]="false" class="c-detailed-info x-no-movie-selected" type="light">
    Select a movie from the list to view its details!
  </ngb-alert>
</ng-template>
