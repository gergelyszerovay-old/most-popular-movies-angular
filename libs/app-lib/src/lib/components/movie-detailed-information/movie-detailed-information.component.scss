$componentClass: '.c-detailed-info';

#{$componentClass} {
  &.x-movie-selected {
    padding: 1rem;
    background-color: white;

    h3 {
      padding-right: 2rem;
    }

    .close {
      display: none;
    }

    #{$componentClass} {
      &--genres {
        margin: 0;
        padding: 0 0 1rem;
        display: block;

        li {
          list-style-type: none;
          display: inline;
          margin-right: 0.5rem;
        }
      }

      &--rating {
        margin-bottom: 1rem;
        font-weight: bold;

        ngb-rating {
          display: inline;
          color: darkgoldenrod;
        }
      }
    }
  }

  &.x-no-movie-selected {
    margin: 1rem;
  }

}

@media (max-width: 767px) {
  #{$componentClass} {
    &.x-no-movie-selected {
      display: none;
    }

    &.x-movie-selected {
      padding: 1rem;
      background-color: white;

      .close {
        display: block;
      }

      #{$componentClass} {
        &--poster-overview {
          img {
            margin-bottom: 1rem;
          }
        }
      }
    }
  }
}


