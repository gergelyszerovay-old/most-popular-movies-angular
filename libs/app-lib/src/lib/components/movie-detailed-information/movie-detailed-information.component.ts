import {ChangeDetectionStrategy, Component, EventEmitter, Input, Output} from '@angular/core';
import {MovieListEntityWithGenreNames} from "../../+state/movie-list/movie-list.models";

@Component({
  selector: 'app-movie-detailed-information',
  templateUrl: './movie-detailed-information.component.html',
  styleUrls: ['./movie-detailed-information.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class MovieDetailedInformationComponent {
  @Input() movie: MovieListEntityWithGenreNames;
  @Input() baseUrlPoster: string;
  @Output() onCloseClick = new EventEmitter<void>();

  constructor() {
  }

  public trackByFnGenreList(index, item) {
    return item.id;
  }
}
