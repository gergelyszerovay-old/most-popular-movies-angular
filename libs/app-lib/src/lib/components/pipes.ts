import {Pipe, PipeTransform} from '@angular/core';

@Pipe({name: 'numberToFixed'})
export class NumberToFixedPipe implements PipeTransform {
  transform(value: number, fractionDigits: number): string {
    return value.toFixed(fractionDigits);
  }
}

@Pipe({name: 'getYearFromISODate'})
export class GetYearFromISODatePipe implements PipeTransform {
  transform(value: string): string {
    return value.substr(0, 4);
  }
}
