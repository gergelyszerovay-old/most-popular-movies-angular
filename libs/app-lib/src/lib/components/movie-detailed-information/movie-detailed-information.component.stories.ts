import {object, text} from '@storybook/addon-knobs';
import {MovieDetailedInformationComponent} from './movie-detailed-information.component';
import {NgbModule} from "@ng-bootstrap/ng-bootstrap";
import {action} from "@storybook/addon-actions";
import {GetYearFromISODatePipe, NumberToFixedPipe} from "../pipes";

export default {
  title: 'MovieDetailedInformationComponent'
}

const moduleMetadata = {
  imports: [NgbModule],
  declarations: [
    NumberToFixedPipe,
    GetYearFromISODatePipe
  ]
};

export const noMovieSelected = () => ({
  moduleMetadata,
  component: MovieDetailedInformationComponent,
  props: {
    movie: object('movie', null),
    baseUrlPoster: text('baseUrlPoster', 'https://image.tmdb.org/t/p/w500/'),
    onCloseClick: action('onCloseClick'),
  }
});

export const movieWithPoster = () => ({
  moduleMetadata,
  component: MovieDetailedInformationComponent,
  props: {
    baseUrlPoster: text('baseUrlPoster', 'https://image.tmdb.org/t/p/w500/'),
    onCloseClick: action('onCloseClick'),
    movie: object('movie', {
      "popularity": 246.384,
      "vote_count": 676,
      "video": false,
      "poster_path": "/bOKjzWDxiDkgEQznhzP4kdeAHNI.jpg",
      "id": 605116,
      "adult": false,
      "backdrop_path": "/qVygtf2vU15L2yKS4Ke44U4oMdD.jpg",
      "original_language": "en",
      "original_title": "Project Power",
      "genre_ids": [
        28,
        80,
        878
      ],
      "title": "Project Power",
      "vote_average": 6.7,
      "overview": "An ex-soldier, a teen and a cop collide in New Orleans as they hunt for the source behind a dangerous new pill that grants users temporary superpowers.",
      "release_date": "2020-08-14",
      "_genres": [{id: 1, name: 'Gernre1'}, {id: 1, name: 'Gernre2'}]
    }),
  }
});

export const movieWithoutPoster = () => ({
  moduleMetadata,
  component: MovieDetailedInformationComponent,
  props: {
    baseUrlPoster: text('baseUrlPoster', 'https://image.tmdb.org/t/p/w500/'),
    onCloseClick: action('onCloseClick'),
    movie: object('movie', {
      "popularity": 246.384,
      "vote_count": 676,
      "video": false,
      "poster_path": null,
      "id": 605116,
      "adult": false,
      "backdrop_path": "/qVygtf2vU15L2yKS4Ke44U4oMdD.jpg",
      "original_language": "en",
      "original_title": "Project Power",
      "genre_ids": [
        28,
        80,
        878
      ],
      "title": "Project Power",
      "vote_average": 6.7,
      "overview": "An ex-soldier, a teen and a cop collide in New Orleans as they hunt for the source behind a dangerous new pill that grants users temporary superpowers.",
      "release_date": "2020-08-14",
      "_genres": [{id: 1, name: 'Gernre1'}, {id: 1, name: 'Gernre2'}]
    }),
  }
});
