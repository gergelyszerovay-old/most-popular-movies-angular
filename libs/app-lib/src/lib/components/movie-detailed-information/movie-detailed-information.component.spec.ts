import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {MovieDetailedInformationComponent} from './movie-detailed-information.component';
import {NgbAlert, NgbRating, NgbToast} from "@ng-bootstrap/ng-bootstrap";

import * as Stories from './movie-detailed-information.component.stories';
import {GetYearFromISODatePipe, NumberToFixedPipe} from "../pipes";
import {MockComponent} from "ng-mocks";

describe('MovieDetailedInformationComponent', () => {
  let component: MovieDetailedInformationComponent;
  let fixture: ComponentFixture<MovieDetailedInformationComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        MovieDetailedInformationComponent,
        NumberToFixedPipe,
        GetYearFromISODatePipe,
        MockComponent(NgbToast),
        MockComponent(NgbRating),
        MockComponent(NgbAlert),
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MovieDetailedInformationComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('trackByFnGenreList should return item.id', () => {
    expect(component.trackByFnGenreList(1, {id: 2})).toBe(2);
  });

  describe('Storybook stories snapshot tests', () => {
    for (const [key, value] of Object.entries(Stories)) {
      if (value instanceof Function) {
        // console.log(`${key}: ${value}`);
        it(`Story ${key}`, () => {
          component = Object.assign(component, Stories[key]().props)
          fixture.detectChanges();
          expect(fixture).toMatchSnapshot();
        });
      }
    }
  });

  // it('Story noMovieSelected', () => {
  //   // console.log(Stories.noMovieSelected().props);
  //   component = Object.assign(component, Stories.noMovieSelected().props)
  //   fixture.detectChanges();
  //
  //   expect(fixture).toMatchSnapshot();
  // });


});
