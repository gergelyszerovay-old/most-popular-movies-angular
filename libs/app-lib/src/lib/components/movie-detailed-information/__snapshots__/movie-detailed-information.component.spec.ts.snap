// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailedInformationComponent Storybook stories snapshot tests Story movieWithPoster 1`] = `
<app-movie-detailed-information
  baseUrlPoster={[Function String]}
  movie={[Function Object]}
  onCloseClick={[Function Function]}
>
  <ngb-alert
    class="c-detailed-info x-no-movie-selected"
    type="light"
  >
     Select a movie from the list to view its details! 
  </ngb-alert>
</app-movie-detailed-information>
`;

exports[`MovieDetailedInformationComponent Storybook stories snapshot tests Story movieWithoutPoster 1`] = `
<app-movie-detailed-information
  baseUrlPoster={[Function String]}
  movie={[Function Object]}
  onCloseClick={[Function Function]}
>
  <ngb-alert
    class="c-detailed-info x-no-movie-selected"
    type="light"
  >
     Select a movie from the list to view its details! 
  </ngb-alert>
</app-movie-detailed-information>
`;

exports[`MovieDetailedInformationComponent Storybook stories snapshot tests Story noMovieSelected 1`] = `
<app-movie-detailed-information
  baseUrlPoster={[Function String]}
  movie="null"
  onCloseClick={[Function Function]}
>
  <ngb-alert
    class="c-detailed-info x-no-movie-selected"
    type="light"
  >
     Select a movie from the list to view its details! 
  </ngb-alert>
</app-movie-detailed-information>
`;
